help_test.go

```diff
@@ -512,3 +512,55 @@ Group 2
 		require.Equal(t, expected, w.String())
 	})
 }
+
+func TestUsageOnError(t *testing.T) {
+	var cli struct {
+		Flag string `help:"A required flag." required`
+	}
+	w := &strings.Builder{}
+	p := mustNew(t, &cli,
+		kong.Writers(w, w),
+		kong.Description("Some description."),
+		kong.Exit(func(int) {}),
+		kong.UsageOnError(),
+	)
+	_, err := p.Parse([]string{"--help"})
+	require.Error(t, err)
+	//p.FatalIfErrorf(err)
+
+	expected := `Usage: test --flag=STRING
+
+Some description.
+
+Flags:
+  -h, --help           Show context-sensitive help.
+      --flag=STRING    A required flag.
+`
+	require.Equal(t, expected, w.String())
+}
+
+func TestShortUsageOnError(t *testing.T) {
+	var cli struct {
+		Flag string `help:"A required flag." required`
+	}
+	w := &strings.Builder{}
+	p := mustNew(t, &cli,
+		kong.Writers(w, w),
+		kong.Description("Some description."),
+		kong.Exit(func(int) {}),
+		kong.ShortUsageOnError(),
+	)
+	_, err := p.Parse([]string{})
+	require.Error(t, err)
+	p.FatalIfErrorf(err)
```

Comment:
Remove?

Reply:
It doesn't print without it (actual output is empty, just double checked).
I followed the flow from `Parse` in `global.go` - is there a better way of doing this?